compiler/frontend/src/org/jetbrains/kotlin/parsing/KotlinParsing.java

```diff
@@ -112,9 +112,20 @@ void parseFile() {
             parseTopLevelDeclaration();
         }
 
+        checkUnclosedBlockComment();
         fileMarker.done(KT_FILE);
     }
 
+    private void checkUnclosedBlockComment() {
+        if (TokenSet.create(BLOCK_COMMENT, DOC_COMMENT).contains(myBuilder.rawLookup(-1))) {
+            int startOffset = myBuilder.rawTokenTypeStart(-1);
+            int endOffset = myBuilder.rawTokenTypeStart(0);
+            if (!myBuilder.getOriginalText().subSequence(startOffset, endOffset).toString().endsWith("*/")) {
```

Comment:
This may materialize a huge file aa a single string (in case of this file being commented entirely)

We could probably use other means, such as:

* changing the lexer to emit a special token type on unfinished comments
* using the CharSequence API to only materialize the last two characters as String, and not the entire text